apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/commons/RetryOnRDSFailoverIT.java

```diff
@@ -0,0 +1,190 @@
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.EnableRetry;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+import software.amazon.jdbc.plugin.failover.FailoverSQLException;
+import software.amazon.jdbc.plugin.failover.FailoverSuccessSQLException;
+
+/**
+ * Spring integration tests for the {@link RetryOnRDSFailover} annotation that extends the {@link
+ * Retryable} annotation from spring-retry.
+ */
+@RunWith(SpringRunner.class)
+@ContextConfiguration(classes = {RetryOnRDSFailoverIT.TestContextConfiguration.class})
+public class RetryOnRDSFailoverIT {
+
+  /**
+   * Autowired bean used in order to activate spring-retry's proxying via the {@link
+   * TestContextConfiguration} configuration.
+   */
+  @Autowired private RetryableOperationBean retryableOperationBean;
+
+  /**
+   * Empty {@link Configuration} that imports {@link RetryableOperationBean} as a bean. Used for
+   * {@link EnableRetry}.
+   */
+  @Configuration
+  @EnableRetry
+  @Import(RetryableOperationBean.class)
+  static class TestContextConfiguration {}
+
+  /**
+   * Bean providing a means for counting retry attempts per-test and a method that is used to test
+   * the {@link Exception}-matching of the {@link RetryOnRDSFailover} annotation.
+   */
+  static class RetryableOperationBean {
+    /**
+     * Map of random GUIDs, which map to a given test case run, to the number of retries for that
+     * given test case. Used to assert on the number of retries per-test-case.
+     */
+    public static final ConcurrentHashMap<String, Integer> RETRY_COUNTS_PER_GUID =
+        new ConcurrentHashMap<>();
+
+    /**
+     * Simple helper method annotated with {@link RetryOnRDSFailover} (with a backoff delay of 1 ms
+     * to speed up tests) that simply throws the {@link Exception}s provided and tracks the number
+     * of times its been retried. Once all {@link Exception}s are exhausted from the provided list,
+     * this method simply returns {@code true}.
+     *
+     * @param exceptions mutable {@link List} of {@link Exception}s which indicate which {@link
+     *     Exception}s to throw prior to executing actual method logic
+     * @param guid random GUID that is used as a key to identify a given test-case run for tracking
+     *     retry counts
+     * @return {@code true}, always
+     * @throws Exception whichever {@link Exception} that is at the top of the provided list of
+     *     {@link Exception}s, if not empty
+     */
+    @RetryOnRDSFailover(backoff = @Backoff(delay = 1))
+    boolean retryableOperation(List<Exception> exceptions, String guid) throws Exception {
+      RETRY_COUNTS_PER_GUID.put(guid, RETRY_COUNTS_PER_GUID.getOrDefault(guid, -1) + 1);
```

Comment:
For anyone curious: I opted to do the counting here manually rather than rely on something like a [`Mockito.spy`](https://site.mockito.org/javadoc/current/org/mockito/Spy.html) to verify call counts of `RetryableOperationBean#retryableOperation` because it seems that `spring-retry` either doesn't call the `@Retryable`-annotated method on retries or `spy` doesn't play nice with the proxied `RetryableOperationsBean` class. Either way, attempting to verifying retry attempts using Mockito's call counting (with `verify()` and `times()`) always returns that the method was called once, even though it _should_ return that it was called multiple times.

So, tracking the number of retry attempts manually was honestly the easiest approach I could think of. A `ConcurrentHashMap` is used in case these tests are ever run in parallel, also.